.github/workflows/release.yml

```diff
@@ -1,20 +1,44 @@
-# This workflow is triggered by new releases
-# It builds, tests, and publishes to the GitHub NuGet package registry
-name: Release package
+# Publish new package versions of ModelContextProtocol
+#
+# Daily and Manual Runs
+# - Triggered automatically at 07:00 UTC daily
+# - Triggered manually using GitHub Actions workflow_dispatch event
+# - Version prefix applied from /src/Directory.Build.props
+# - Version suffix set to `ci.{github.run_number}`
+# - Package published to GitHub package registry
+#
+# Official Releases
+# - Triggered after a GitHub Release is created
+# - Version prefix applied from /src/Directory.Build.props
+# - Version suffix applied from /src/Directory.Build.props
+# - Package published to GitHub package registry
+# - Package published to NuGet.org
+# - Version prefix and/or suffix should be updated after each release
+
+name: Release Publishing
 
 on:
+  schedule:
+    - cron: '0 7 * * *'
+
+  workflow_dispatch:
+
   release:
     types: [published]
 
 jobs:
-  build:
+  run-ci:
+    uses: ./.github/workflows/ci.yml
 
+  build-package:
     runs-on: windows-latest
+    needs: run-ci
```

Comment:
Does this mean that all steps in ci.yml will be run? I'm guessing it's going to pick `windows-latest` from the matrix but it isn't clear to me if it will be using `Debug` or `Release` configuration.

Reply:
Should we be blocking the release pipeline on tests? We're currently dealing with flakiness in integration testing so this could be a problem.

Reply:
This will run the entire ci pipeline across all configs, and that does indeed need to be green for the publishing to proceed. During testing, I saw the flakiness show up and cause this job to fail as you suspected. The job can be re-run if it fails, which is reasonable human intervention to force a new package to be published despite CI being red.

We could also pass an argument to the CI run from this job to skip some/all of the tests while still ensuring the build is successful across the whole matrix.

Reply:
I updated this to no longer call the full CI workflow. Instead, I've copied the _build_ matrix from the CI workflow into this one. So as long as the build succeeds in all configs, the release will proceed.